### ISSUE TITLE
Compression Result

### ISSUE THREAD

**Issue by marifnst** [yuanyuanli85/CaffeModelCompression#2]
Hello

I'm so interested for your project.
I have tried your code for VGG16 net, but why compress output become npz ?
Can i use this npz output to be pre trained model for my caffe ?

thanks

**Comment by NicolaiHarich** [comment]
I have the same question. So far I've testet the decompressed model, which has the same size, but lost 15 per cent accuracy compared to the original weights. I guess a final finetuning is needed anyway, but how can I use the compressed net?
- Nico

**Comment by yuanyuanli85** [comment]
@marifnst 

caffe only supports fp32 and double, does not support another type, like uint8 and fp16.  So, we have to retrieve the compressed model to fp32 or double when you load the weights.

**Comment by yuanyuanli85** [comment]
@NicolaiHarich 15% accuracy loss is too huge.  We have tried vgg16 and alexnet, only 1% loss in test accuracy (fc'weights to 16bits and conv's weights to 16bits).  Fine tuning is required if you don't want to loss any accuracy.

**Comment by marifnst** [comment]
thanks @yuanyuanli85 
any sample to convert compressed model to fp32 ?
i'm still search how to implement your solution.

thanks

**Comment by marifnst** [comment]
hi @yuanyuanli85 

after search and try many time, i can compress my caffemodel.
this is link of my sample code [SampleCaffeModelCompression](https://github.com/marifnst/SampleCaffeModelCompression)
thank you very much for your code.

thanks

**Comment by yuanyuanli85** [comment]
@marifnst  in this repo, the decompress funciton "caffe_model_decompress" is also provided. That decompress the model into fp32 which can be used by caffe.

**Comment by marifnst** [comment]
@yuanyuanli85 okay, thank you very much for your code.
it really help me a lot.

best regards

**Comment by chenusc11** [comment]
@yuanyuanli85 Hi, thanks for your code. I'm wondering when you said "We have tried vgg16 and alexnet, only 1% loss in test accuracy (fc'weights to 16bits and conv's weights to 16bits)." Did it involve retraining?

**Comment by wenston2006** [comment]
After compression and decompression, I found that the face recognition accuracy(AUC) over LFW was raised from 95% to 99%? Is that possible? Does compression lead to overfitting over some testing dataset? But I have not do anything except compression.
